### PR TITLE
Read all interfaces (IPv4 and IPv6) from getifaddrs

### PR DESCRIPTION
**Pull request by asias91**
Netlink calls removed for reading IPv4 interfaces.
Unification and refactoring of code regarding reading interfaces and detecting if address is external.
Special case for bridges removed.

### CODE REVIEW

File: libservice/headers/service/InterfacesReader.h [hunk 1]
```diff
@@ -18,38 +18,41 @@
...
-using BridgeIndices = std::vector<int>;
 
-class NetlinkCalls {
+class InterfacesReader {
 public:
-	virtual ~NetlinkCalls() = default;
+	virtual ~InterfacesReader() = default;
+	void printNetworkInterfacesInfo();
+	[[nodiscard]] virtual std::vector<Ipv4Network> getIpV4Interfaces() const;
+	[[nodiscard]] virtual std::vector<Ipv6Network> getIpV6Interfaces() const;
```
Comment (pawsten) [inline libservice/headers/service/InterfacesReader.h:48]: any reason for [[nondiscard]] ?

File: libservice/src/InterfacesReader.cpp [hunk 2]
```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2023 Dynatrace LLC
```
[the remark this answers is not in the document]
Comment (asias91) [reply libservice/src/InterfacesReader.cpp:2]: I see no need in updating such things. It's refactored (and renamed) file, not a new one.

File: libservice/headers/service/IpAddressChecker.h [hunk 3]
```diff
@@ -28,8 +29,8 @@ class IpAddressChecker {
 public:
 	virtual ~IpAddressChecker() = default;
 
-	virtual bool isV4AddressExternal(IPv4int addr) const = 0;
+	[[nodiscard]] virtual bool isV4AddressExternal(in_addr_t addr) const = 0;
```
[the remark this answers is not in the document]
Comment (asias91) [reply libservice/headers/service/IpAddressChecker.h]: Mainly it was clang-tidy suggestion in my IDE. In general, there are different approaches to nodiscard usage. IMO additional warning about undesired method call is rather beneficial.

File: libservice/src/InterfacesReader.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (kamiljoner) [reply libservice/src/InterfacesReader.cpp:2]: Confusion somes from the fact that change was not-trivial (at least for github) and it did not pickup that it is rename from NetlinkCalls. I agree that there is not much point in updating it to 2024. Even if then, 2023-2024 makes more sense than pure 2024

**Review by pawsten — APPROVED** [review]

File: libservice/headers/service/IpAddressChecker.h (hunk 3 above)
[the remark this answers is not in the document]
Comment (asias91) [reply libservice/headers/service/IpAddressChecker.h]: We can establish sth like coding convention and discuss it separately. As for now, I'm leaving it as it is.

**Pull request merged**